src/Microsoft.AspNetCore.Hosting/Internal/WebHostOptions.cs

```diff
@@ -22,6 +22,7 @@ public WebHostOptions(IConfiguration configuration)
             }
 
             Application = configuration[WebHostDefaults.ApplicationKey];
+            EnvironmentApplication = configuration[WebHostDefaults.EnvironmentApplicationKey];
```

Comment:
When running functional tests, we create a new `IApplicationEnvironment` and set the `ApplicationName` and `ApplicationBasePath` to the the website being run. See https://github.com/aspnet/Mvc/blob/dev/test/Microsoft.AspNetCore.Mvc.FunctionalTests/MvcTestFixture.cs#L70